rdfproxy/utils/sparql_utils.py

```diff
@@ -20,21 +20,21 @@
 """)
 
 
-def replace_query_select_clause(query: str, repl: str) -> str:
+def replace_query_select_clause(query: str, repl: str):
     """Replace the SELECT clause of a query with repl."""
-    if re.search(r"select\s.+", query, re.I) is None:
+    pattern = r"(select\s+[\s\S]*?)(?=\s+where)"
+
+    if re.search(pattern=pattern, string=query, flags=re.IGNORECASE) is None:
```

Comment:
I would prefer to have the stylistic changes in a separate PR. I think it is oke to move the pattern to `pattern =`, but leave the other changes for another time

Reply:
Apart from fixing and assigning the regex pattern, I would see those changes more as a refactoring.

`re.I` is synonymous with `re.IGNORECASE`, kwargs are used instead of positional args and `re.sub` is returned directly instead of binding and returning its result.

Reply:
If you want me to, I can open a separate PR though, no prob.